LayoutTests/inspector/console/setConsoleClearAPIEnabled.html

```diff
@@ -0,0 +1,111 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="../../http/tests/inspector/resources/inspector-test.js"></script>
+<script>
+function test()
+{
+    let suite = InspectorTest.createAsyncSuite("Console.setConsoleClearAPIEnabled");
+
+    suite.addTestCase({
+        name: "Console.setConsoleClearAPIEnabled.API.Disabled",
+        description: "Check that console.clear() doesn't clear the console when disallowed.",
+        async test() {
+            await ConsoleAgent.setConsoleClearAPIEnabled(false);
+
+            let cleared = false;
+            let clearedListener = WI.consoleManager.addEventListener(WI.ConsoleManager.Event.Cleared, (event) => {
+                cleared = true;
+            });
+
+            let messageAdded = false;
+            let messageAddedListener = WI.consoleManager.addEventListener(WI.ConsoleManager.Event.MessageAdded, (event) => {
+                messageAdded = true;
+            });
+
+            await InspectorTest.evaluateInPage(`console.clear()`);
+
+            InspectorTest.expectFalse(cleared, "Should not clear console.");
+            InspectorTest.expectFalse(messageAdded, "Should not send a message.");
```

Comment:
What guarantees that the event handlers on the frontend _did not run_ and that's why the flags both `false`?

Reply:
oh hmm good point.  ill add a `console.log(42);` or something after the `console.clear()` to make sure